### ISSUE TITLE
Feature Request: set TSS log level per logging module

### ISSUE THREAD

**Issue by joholl** [tpm2-software/tpm2-pytss#49]
Unfortunately, there seems to be no mechanism to set the TSS log level at runtime besides via the env variable `TSS2_LOG`, e.g. `TSS2_LOG=all+ERROR,fapi+TRACE`, as documented [here](https://github.com/tpm2-software/tpm2-tss/blob/master/doc/logging.md). What do you think of having a function setting this variable, maybe from a dict?

Example:

```python
log_levels = { "all": "error", "tcti": "trace"}
log_levels_str = ",".join(f"{k}+{v}" for k, v in log_levels.items())  # 'all+error,tcti+trace'
os.environ["TSS2_LOG"] = log_levels_str
```

We might want to use enums instead of strings(?). In case you wonder, modules and log levels are parsed **case insensitive**.

Complete list of TSS log levels (as defined [here](https://github.com/tpm2-software/tpm2-tss/blob/master/src/util/log.h#L39)):
* none
* (unused)  [note: defined with parens]
* ERROR
* WARNING
* info
* debug
* trace

Complete list of TSS log modules (grepped):
* fapi
* esys
* esys_crypto
* sys
* marshal
* tcti
* log
* test
* all  [note: covers all modules]

**Comment by pdxjohnny** [comment]
Sweet. I'm doing a major re-write at the moment. I'll make sure to get this in there. Thanks!

**Comment by williamcroberts** [comment]
We don;t have a log in the rewrite to CFFI yet, but we will add it.

**Comment by joholl** [comment]
Nice, I did some preliminary work on this. That includes a PR for redirecting logging to a file (which could be a pipe) on TSS side,  see tpm2-software/tpm2-tss#1976

**Comment by williamcroberts** [comment]
I don't think this is going to make 1.0 and we need to finalize the tss C code approach. We can't do per-ctx logging because much of the submodules don't have context parameters, like libmu.

See:
  - https://github.com/tpm2-software/tpm2-tss/pull/2087
